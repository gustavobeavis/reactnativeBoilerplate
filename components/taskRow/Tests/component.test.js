import React from 'react';
import renderer from 'react-test-renderer';
import TaskRow from '../component';

test('TaskRow renders correctly', () => {
  // Arrange
  const todo = { task: 'exemplo' };

  // Act
  const tree = renderer
    .create(<TaskRow todo={todo} removeTodo={jest.fn} />)
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});
