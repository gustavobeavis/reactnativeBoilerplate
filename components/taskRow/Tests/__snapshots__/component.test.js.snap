// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskRow renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEnabled={false}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  viewIsInsideTabBar={false}
  virtual={undefined}
>
  <View>
    <View
      style={
        Array [
          Object {},
          Object {
            "borderColor": "#fff",
            "borderWidth": 0.5,
            "marginBottom": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 0,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": null,
                "alignSelf": null,
                "flex": 0,
              },
              Object {
                "width": 0,
                "zIndex": 1,
              },
            ]
          }
          virtual={undefined}
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "flex": 0,
              },
            ]
          }
          virtual={undefined}
        />
        <View
          style={
            Array [
              Object {
                "alignItems": null,
                "alignSelf": null,
                "flex": 0,
              },
              Object {
                "width": 100,
                "zIndex": 1,
              },
            ]
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#05A5D1",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 0,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 2,
                "flex": 1,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "marginBottom": 10,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              active={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 24,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "paddingTop": 2,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              virtual={undefined}
            >
              ï“…
            </Text>
          </View>
        </View>
      </View>
      <View
        collapsable={undefined}
        onLayout={[Function]}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 2,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            list={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 0,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                },
                Object {},
                Object {
                  "borderColor": "#fff",
                  "borderWidth": 0.5,
                  "marginBottom": 10,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#fff",
                    "borderColor": "#E7E7E7",
                    "borderWidth": 1,
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 60,
                    "justifyContent": "space-between",
                    "marginBottom": 10,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "padding": 10,
                  },
                  Object {
                    "borderWidth": 0,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 22,
                    "fontWeight": "300",
                    "width": "100%",
                  }
                }
              >
                exemplo
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
